base/dict.jl

```diff
@@ -361,7 +362,13 @@ function ht_keyindex2!(h::Dict{K,V}, key) where V where K
 
     avail < 0 && return avail
 
-    maxallowed = max(maxallowedprobe, sz>>maxprobeshift)
+    if hashval == 0
```

Comment:
Does this work ok if hash==0 and other "properly" hashed types are mixed?

For instance if a "properly" hashed instance has `index==1` (say `sz==hashval`), shouldn't then the following loop run over `maxallowed = sz - 1`.  A similar problem could be encountered if `index==2` and many `hashval==0` are present.

Reply:
Yes you're right, I'll fix this.